DELETE statement from executed in MySQL doesn't executed in starrocks.

When I delete rows in MySQL, the data still present in StarRocks.

Flink version: 1.15
MySQL CDC version: 2.2.1
StarRocks connector version: 1.2.4

Hi @banmoy. Using `'sink.properties.format' =' csv'` caused another issue:
```
Caused by: java.lang.RuntimeException: com.starrocks.data.load.stream.exception.StreamLoadFailException: {
    "TxnId": 46420,
    "Label": "ff11b302-f5dd-4886-962b-1832df9ba4ae",
    "Status": "Fail",
    "Message": "Syntax error in line 1:\n...tid,refid,templateid,type,create,edit,view,created_by,...\n                             ^\nEncountered: CREATE\nExpected: CREATE is keyword, maybe `CREATE`\n",
    "NumberTotalRows": 0,
    "NumberLoadedRows": 0,
    "NumberFilteredRows": 0,
    "NumberUnselectedRows": 0,
    "LoadBytes": 0,
    "LoadTimeMs": 0,
    "BeginTxnTimeMs": 0,
    "StreamLoadPlanTimeMs": 1,
    "ReadDataTimeMs": 0,
    "WriteDataTimeMs": 0,
    "CommitAndPublishTimeMs": 0
}
```

More log:
```
2022-12-09 02:54:24,863 INFO  org.apache.flink.runtime.executiongraph.ExecutionGraph       [] - Source: table_1_src[97] -> ConstraintEnforcer[98] -> Sink: table_1_sink[98] (1/1) (8a598b4550a1f3c2bc87451351ff8da1) switched from RUNNING to FAILED on basic-session-taskmanager-1-1 @xxx.compute.internal (dataPort=33177).
java.lang.RuntimeException: com.starrocks.data.load.stream.exception.StreamLoadFailException: {
    "TxnId": 46420,
    "Label": "ff11b302-f5dd-4886-962b-1832df9ba4ae",
    "Status": "Fail",
    "Message": "Syntax error in line 1:\n...tid,refid,templateid,type,create,edit,view,created_by,...\n                             ^\nEncountered: CREATE\nExpected: CREATE is keyword, maybe `CREATE`\n",
    "NumberTotalRows": 0,
    "NumberLoadedRows": 0,
    "NumberFilteredRows": 0,
    "NumberUnselectedRows": 0,
    "LoadBytes": 0,
    "LoadTimeMs": 0,
    "BeginTxnTimeMs": 0,
    "StreamLoadPlanTimeMs": 1,
    "ReadDataTimeMs": 0,
    "WriteDataTimeMs": 0,
    "CommitAndPublishTimeMs": 0
}
	at com.starrocks.data.load.stream.DefaultStreamLoadManager.AssertNotException(DefaultStreamLoadManager.java:333) ~[?:?]
	at com.starrocks.data.load.stream.DefaultStreamLoadManager.write(DefaultStreamLoadManager.java:195) ~[?:?]
	at com.starrocks.connector.flink.table.sink.StarRocksDynamicSinkFunctionV2.invoke(StarRocksDynamicSinkFunctionV2.java:152) ~[?:?]
	at org.apache.flink.table.runtime.operators.sink.SinkOperator.processElement(SinkOperator.java:65) ~[flink-table-runtime-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.pushToOperator(CopyingChainingOutput.java:82) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:57) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:29) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.api.operators.CountingOutput.collect(CountingOutput.java:56) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.api.operators.CountingOutput.collect(CountingOutput.java:29) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.table.runtime.util.StreamRecordCollector.collect(StreamRecordCollector.java:44) ~[flink-table-runtime-1.15.3.jar:1.15.3]
	at org.apache.flink.table.runtime.operators.sink.ConstraintEnforcer.processElement(ConstraintEnforcer.java:247) ~[flink-table-runtime-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.pushToOperator(CopyingChainingOutput.java:82) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:57) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:29) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.SourceOperatorStreamTask$AsyncDataOutputToOutput.emitRecord(SourceOperatorStreamTask.java:313) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.api.operators.source.SourceOutputWithWatermarks.collect(SourceOutputWithWatermarks.java:110) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.api.operators.source.SourceOutputWithWatermarks.collect(SourceOutputWithWatermarks.java:101) ~[flink-dist-1.15.3.jar:1.15.3]
	at com.ververica.cdc.connectors.mysql.source.reader.MySqlRecordEmitter$OutputCollector.collect(MySqlRecordEmitter.java:143) ~[?:?]
	at com.ververica.cdc.debezium.table.RowDataDebeziumDeserializeSchema.emit(RowDataDebeziumDeserializeSchema.java:157) ~[?:?]
	at com.ververica.cdc.debezium.table.RowDataDebeziumDeserializeSchema.deserialize(RowDataDebeziumDeserializeSchema.java:124) ~[?:?]
	at com.ververica.cdc.connectors.mysql.source.reader.MySqlRecordEmitter.emitElement(MySqlRecordEmitter.java:118) ~[?:?]
	at com.ververica.cdc.connectors.mysql.source.reader.MySqlRecordEmitter.emitRecord(MySqlRecordEmitter.java:100) ~[?:?]
	at com.ververica.cdc.connectors.mysql.source.reader.MySqlRecordEmitter.emitRecord(MySqlRecordEmitter.java:54) ~[?:?]
	at org.apache.flink.connector.base.source.reader.SourceReaderBase.pollNext(SourceReaderBase.java:143) ~[flink-connector-files-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.api.operators.SourceOperator.emitNext(SourceOperator.java:385) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.io.StreamTaskSourceInput.emitNext(StreamTaskSourceInput.java:68) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.io.StreamOneInputProcessor.processInput(StreamOneInputProcessor.java:65) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.processInput(StreamTask.java:519) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMailboxLoop(MailboxProcessor.java:203) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.runMailboxLoop(StreamTask.java:807) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:756) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.runtime.taskmanager.Task.runWithSystemExitMonitoring(Task.java:948) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.runtime.taskmanager.Task.restoreAndInvoke(Task.java:927) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:741) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:563) ~[flink-dist-1.15.3.jar:1.15.3]
	at java.lang.Thread.run(Unknown Source) ~[?:?]
Caused by: com.starrocks.data.load.stream.exception.StreamLoadFailException: {
    "TxnId": 46420,
    "Label": "ff11b302-f5dd-4886-962b-1832df9ba4ae",
    "Status": "Fail",
    "Message": "Syntax error in line 1:\n...tid,refid,templateid,type,create,edit,view,created_by,...\n                             ^\nEncountered: CREATE\nExpected: CREATE is keyword, maybe `CREATE`\n",
    "NumberTotalRows": 0,
    "NumberLoadedRows": 0,
    "NumberFilteredRows": 0,
    "NumberUnselectedRows": 0,
    "LoadBytes": 0,
    "LoadTimeMs": 0,
    "BeginTxnTimeMs": 0,
    "StreamLoadPlanTimeMs": 1,
    "ReadDataTimeMs": 0,
    "WriteDataTimeMs": 0,
    "CommitAndPublishTimeMs": 0
}
	at com.starrocks.data.load.stream.DefaultStreamLoader.send(DefaultStreamLoader.java:270) ~[?:?]
	at com.starrocks.data.load.stream.DefaultStreamLoader.lambda$send$2(DefaultStreamLoader.java:113) ~[?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
	... 1 more
2022-12-09 02:54:24,883 INFO  org.apache.flink.runtime.source.coordinator.SourceCoordinator [] - Removing registered reader after failure for subtask 0 of source Source: table_1_src[97].
```

SHOW CREATE TABLE in MySQL
```
CREATE TABLE `table_1` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `parentid` int(11) DEFAULT NULL COMMENT 'parentid of settings',
  `refid` int(11) DEFAULT NULL COMMENT 'profileid or deptid',
  `templateid` varchar(255) DEFAULT NULL COMMENT 'templateid for settings',
  `type` varchar(255) DEFAULT NULL COMMENT '''user'' or ''department''',
  `create` int(11) DEFAULT '1' COMMENT '0-default,1-allow,2-create',
  `edit` int(11) DEFAULT '1' COMMENT '0-default,1-allow,2-create',
  `view` int(11) DEFAULT '1' COMMENT '0-default,1-allow,2-create',
  `created_by` int(11) DEFAULT NULL COMMENT 'profileid created',
  `modified_by` int(11) DEFAULT NULL COMMENT 'last profileid modified',
  `created_datetime` datetime DEFAULT NULL COMMENT 'settings created date',
  `modified_datetime` datetime DEFAULT NULL COMMENT 'settings modified date'
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=11194 DEFAULT CHARSET=latin1
```

SHOW CREATE TABLE in StarRocks:
```
CREATE TABLE `table_1` (
  `id` int(11) NOT NULL COMMENT "",
  `parentid` int(11) NULL COMMENT "parentid of settings",
  `refid` int(11) NULL COMMENT "profileid or deptid",
  `templateid` varchar(65533) NULL COMMENT "templateid for settings",
  `type` varchar(65533) NULL COMMENT "'user' or 'department'",
  `create` int(11) NULL DEFAULT "1" COMMENT "0-default,1-allow,2-create",
  `edit` int(11) NULL DEFAULT "1" COMMENT "0-default,1-allow,2-create",
  `view` int(11) NULL DEFAULT "1" COMMENT "0-default,1-allow,2-create",
  `created_by` int(11) NULL COMMENT "profileid created",
  `modified_by` int(11) NULL COMMENT "last profileid modified",
  `created_datetime` datetime NULL COMMENT "settings created date",
  `modified_datetime` datetime NULL COMMENT "settings modified date"
) ENGINE=OLAP 
PRIMARY KEY(`id`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`id`) BUCKETS 1 
PROPERTIES (
"replication_num" = "3",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "false"
);
```

--- a/src/main/java/com/starrocks/connector/flink/table/sink/StarRocksDynamicSinkFunctionV2.java
+++ b/src/main/java/com/starrocks/connector/flink/table/sink/StarRocksDynamicSinkFunctionV2.java
@@ -65,11 +65,13 @@ public class StarRocksDynamicSinkFunctionV2<T> extends StarRocksDynamicSinkFunct
         this.sinkOptions = sinkOptions;
         this.rowTransformer = rowTransformer;
         rowTransformer.setTableSchema(schema);
-        this.serializer = StarRocksSerializerFactory.createSerializer(sinkOptions, schema.getFieldNames());
         StarRocksSinkTable sinkTable = StarRocksSinkTable.builder()
                 .sinkOptions(sinkOptions)
                 .build();
         sinkTable.validateTableStructure(sinkOptions, schema);
+        // StarRocksJsonSerializer depends on SinkOptions#supportUpsertDelete which is decided in
+        // StarRocksSinkTable#validateTableStructure, so create serializer after validating table structure
+        this.serializer = StarRocksSerializerFactory.createSerializer(sinkOptions, schema.getFieldNames());
         this.sinkManager = new StarRocksSinkManagerV2(sinkOptions.getProperties());
     }
 

--- a/starrocks-stream-load-sdk/src/main/java/com/starrocks/data/load/stream/DefaultStreamLoadManager.java
+++ b/starrocks-stream-load-sdk/src/main/java/com/starrocks/data/load/stream/DefaultStreamLoadManager.java
@@ -1,9 +1,8 @@
 package com.starrocks.data.load.stream;
 
+import com.alibaba.fastjson.JSON;
 import com.starrocks.data.load.stream.properties.StreamLoadProperties;
 import com.starrocks.data.load.stream.properties.StreamLoadTableProperties;
-
-import com.alibaba.fastjson.JSON;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -26,6 +25,8 @@ import java.util.concurrent.locks.LockSupport;
 import java.util.concurrent.locks.ReentrantLock;
 
 public class DefaultStreamLoadManager implements StreamLoadManager, Serializable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultStreamLoadManager.class);
 
     enum State {
         ACTIVE,
@@ -193,6 +194,10 @@ public class DefaultStreamLoadManager implements StreamLoadManager, Serializable
         TableRegion region = getCacheRegion(uniqueKey, database, table);
         for (String row : rows) {
             AssertNotException();
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Write uniqueKey {}, database {}, table {}, row {}",
+                        uniqueKey == null ? "null" : uniqueKey, database, table, row);
+            }
             int bytes = region.write(row.getBytes(StandardCharsets.UTF_8));
             if (currentCacheBytes.addAndGet(bytes) >= maxCacheBytes) {
                 int idx = 0;

